### ISSUE TITLE
Movies will not be grabbed with negative availability 

### ISSUE THREAD

**Issue by samwiseg00** [Radarr/Radarr#1233]
**Description:**
It seems that movies cannot be downloaded when there is a min availability set to a negative number. There are several instances in the log where there is an error on a movie that has been available for months or even years. 

My current setting is -60 (which I assume is 60 days before availability)

**Radarr Version:**
0.2.0.535

**Logs:**
```
Couldn't evaluate decision on XXXXXXXXXX.2016.1080p.BluRay.DTS-HD.MA.5.1.x264-FuzerHD English, with spec: AvailabilitySpecification: The added or subtracted value results in an un-representable DateTime.
Parameter name: value

System.ArgumentOutOfRangeException: The added or subtracted value results in an un-representable DateTime.
Parameter name: value
  at System.DateTime.AddTicks (System.Int64 value) [0x00035] in /builddir/build/BUILD/mono-4.6.2/mcs/class/referencesource/mscorlib/system/datetime.cs:457 
  at System.DateTime.Add (System.Double value, System.Int32 scale) [0x0007b] in /builddir/build/BUILD/mono-4.6.2/mcs/class/referencesource/mscorlib/system/datetime.cs:355 
  at System.DateTime.AddDays (System.Double value) [0x00000] in /builddir/build/BUILD/mono-4.6.2/mcs/class/referencesource/mscorlib/system/datetime.cs:365 
  at NzbDrone.Core.Tv.Movie.IsAvailable (System.Int32 delay) [0x000c1] in C:\projects\radarr-usby1\src\NzbDrone.Core\Tv\Movie.cs:101 
  at NzbDrone.Core.DecisionEngine.Specifications.RssSync.AvailabilitySpecification.IsSatisfiedBy (NzbDrone.Core.Parser.Model.RemoteMovie subject, NzbDrone.Core.IndexerSearch.Definitions.SearchCriteriaBase searchCriteria) [0x00021] in C:\projects\radarr-usby1\src\NzbDrone.Core\DecisionEngine\Specifications\RssSync\AvailabilitySpecification.cs:34 
  at NzbDrone.Core.DecisionEngine.DownloadDecisionMaker.EvaluateSpec (NzbDrone.Core.DecisionEngine.IDecisionEngineSpecification spec, NzbDrone.Core.Parser.Model.RemoteMovie remoteMovie, NzbDrone.Core.IndexerSearch.Definitions.SearchCriteriaBase searchCriteriaBase) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\DecisionEngine\DownloadDecisionMaker.cs:248

```

**Comment by galli-leo** [comment]
Which movie is this? Also does this happen for every movie? Also what availability do you have set?

**Comment by samwiseg00** [comment]
For example this movie. it is set to preDB. 

Does it have to be set to Physical/Web?

```
Couldn't evaluate decision on Moana-2016-1080p-BluRay-x264-DTS-JYK[Sky torrents], with spec: AvailabilitySpecification: The added or subtracted value results in an un-representable DateTime.
Parameter name: value

System.ArgumentOutOfRangeException: The added or subtracted value results in an un-representable DateTime.
Parameter name: value
  at System.DateTime.AddTicks (System.Int64 value) [0x00035] in /builddir/build/BUILD/mono-4.6.2/mcs/class/referencesource/mscorlib/system/datetime.cs:457 
  at System.DateTime.Add (System.Double value, System.Int32 scale) [0x0007b] in /builddir/build/BUILD/mono-4.6.2/mcs/class/referencesource/mscorlib/system/datetime.cs:355 
  at System.DateTime.AddDays (System.Double value) [0x00000] in /builddir/build/BUILD/mono-4.6.2/mcs/class/referencesource/mscorlib/system/datetime.cs:365 
  at NzbDrone.Core.Tv.Movie.IsAvailable (System.Int32 delay) [0x000c1] in C:\projects\radarr-usby1\src\NzbDrone.Core\Tv\Movie.cs:101 
  at NzbDrone.Core.DecisionEngine.Specifications.RssSync.AvailabilitySpecification.IsSatisfiedBy (NzbDrone.Core.Parser.Model.RemoteMovie subject, NzbDrone.Core.IndexerSearch.Definitions.SearchCriteriaBase searchCriteria) [0x00021] in C:\projects\radarr-usby1\src\NzbDrone.Core\DecisionEngine\Specifications\RssSync\AvailabilitySpecification.cs:34 
  at NzbDrone.Core.DecisionEngine.DownloadDecisionMaker.EvaluateSpec (NzbDrone.Core.DecisionEngine.IDecisionEngineSpecification spec, NzbDrone.Core.Parser.Model.RemoteMovie remoteMovie, NzbDrone.Core.IndexerSearch.Definitions.SearchCriteriaBase searchCriteriaBase) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\DecisionEngine\DownloadDecisionMaker.cs:248 
```

**Comment by galli-leo** [comment]
No that shouldn't be necessary. Can you update the info for this movie?

**Comment by SWu** [comment]
Actually, how does preDB work? If a movie doesn't have a scene release yet, how does min availability: preDB know what the actual date is?

As a feature request, would it be possible to also enforce desired quality level to match the preDB entry? E.g., just because a dvdscr has been released, don't mark a movie that has desired quality blu-ray as being available.

**Comment by janipewter** [comment]
It would be helpful to know how the PreDB min availability works. As an example I had Rogue One in my wanted list, using PreDB min avail. There was a scene release of it on 18/03 however even until today, Radarr will not touch it. I pressed the manual search button and it shows the release, and it matches all of my filters and Radarr does not show any reason why it wouldn't be grabbed. But it won't touch it.

I deleted it and re-added using the add+search button and nothing. So I deleted it and re-added but this time using Announced as min availability and it grabbed it straight away.

https://predb.me/?post=6270360

I've now changed all of my monitored movies to In Cinemas min availability until this is resolved.
